### ISSUE TITLE
Ability to pass to a pointer of the logging file already created

### ISSUE THREAD

**Issue by bishtgautam** [RDycore/RDycore#75]
For an E3SM-RDycore run, E3SM will open a logging file named `rdy.log.JOBID` (e.g. see `iulog` in https://github.com/RDycore/E3SM/commit/6f9963c5c302e7e86aad3d66def1eab6816412af#diff-f561dda7e4923546b71863e7959473d32a0668271a8ffe2b410ede9ede15e1c9R50). We need the ability to pass to RDycore `iulog`, so it can be set in `rdy->log` during [`RDySetup()`](https://github.com/RDycore/RDycore/blob/main/src/rdysetup.c#L931-L935)

**Comment by jeff-cohere** [comment]
Unfortunately, Fortran has its own peculiar way of handling I/O, and there's no portable way to pass a Fortran file unit to C for use as a file descriptor. If we want to have this capability, we have 2 options:

1. Create a completely separate subsystem for logging messages to a Fortran file unit (implemented in Fortran and **called by RDycore by all log-related functions**, if and only if a Fortran file unit was specified as a log destination)
2. Create an interface that opens a C file descriptor from Fortran and passes that along to RDycore to use standard C I/O.

Option 1 would allow us to support code like that you've linked to, but would require a lot of data to be passed back and forth between C and Fortran, and would likely be difficult to implement correctly and support.

Option 2 would be much simpler, since it would allow RDycore to control file I/O in a much more standard and flexible way, but it would come with some caveats:
* You'd call something like `rdycore_setio` with a filename instead of a file unit number, which would open a C file descriptor and allow RDycore to write to it
* You would have to call some sort of wrapper function to write to the C file descriptor using RDycore's interface. The easiest thing I can think of is a simplified version of [RDyLog](https://github.com/RDycore/RDycore/blob/main/include/private/rdylogimpl.h#L30) and [its variants](https://github.com/RDycore/RDycore/blob/main/include/private/rdylogimpl.h#L38). E.g.

```
  character(len=1024) :: log_file
  ...
  call RDyOpenLog(log_file)
  ...
  call RDyLogDebug("This is a log debug message.")
```
But Option 2 also has the benefit that it would automatically write log information only to rank 0 and couldn't be manipulated in ways that are incompatible with our existing I/O mechanism.

In other words, Option 1 is the most "Fortran-like" way of doing I/O, but requires us to have a completely separate I/O mechanism under the hood in addition to the one we're already using. Option 2 allows us to use our existing I/O mechanism (and in my opinion is "the right way" to do it), but makes the E3SM Fortran code look a bit more like what we do in C.

The important thing here is that we do all I/O the same way--either all in C or all in Fortran--for a given simulation, because Fortran I/O is completely incompatible with C I/O.

**Comment by bishtgautam** [comment]
I'm onboard with Option-2. Thanks.
